### ISSUE TITLE
Agent IDs starting with a “-“ are not valid devices

### ISSUE THREAD

**Issue by deldrid1** [EatonGMBD/imp-central-impt#8]
Looks like there is a bug in the way the options parsing works – there is no way to use an Agent ID that starts with a ‘-‘…

```shell
$ impt device assign -d "-ZDervOZyQbL" -g 7c4b5300-282e-0eba-3d08-854251674a23 -q;

Usage: impt device assign [--account <account_id>] --device <DEVICE_IDENTIFIER>
[--dg <DEVICE_GROUP_IDENTIFIER>] [--confirmed] [--output <mode>] [--help]

Assigns the specified device to the specified Device Group. Fails if the
specified Device Group does not exist.

Options:
  --help, -h       Displays a description of the command. Ignores any other
                   options.                                            [boolean]
  --account, --ac  The authenticted account identifier: an account ID.  [string]
  --device, -d     A Device identifier: Device ID, MAC address, Agent ID or
                   Device name.                              [string] [required]
  --dg, -g         A Device Group identifier: Device Group ID or Device Group
                   name. If not specified, the Device Group referenced by the
                   Project file in the current directory is used (if there is no
                   Project file, the command fails).                    [string]
  --confirmed, -q  Executes the operation without asking additional confirmation
                   from user.                                          [boolean]
  --output, -z     Adjusts the command's output.
                                  [string] [choices: "minimal", "json", "debug"]
Error: Not enough arguments following: d
IMPT COMMAND FAILS
```

**Comment by alexey-nbl** [comment]
This is an issue (most probably a bug) in yargs-parser module which is used by impt.
We are trying to investigate and find a safe workaround.

**Comment by alexey-nbl** [comment]
Investigated deeply.

The issue affects the both EatonGMBD/imp-central-impt and electricimp/imp-central-impt

Found out the root cause of the issue - regression in yargs-parser module ("nargs" option processing).
A bug against yargs-parser module submitted - https://github.com/yargs/yargs-parser/issues/232

Two possible workarounds determined and investigated:

1) Downgrade yargs module used by impt to the version 10.0.3 (it includes yargs-parser w/o the regression).
This workaround is thoroughly tested, no issues found, minor update in the tests is needed. But it sticks impt to an old version of yargs.
If/when the bug in yargs-parser is fixed, then return to use the latest yargs versions.

2) Continue to use the latest versions of yargs (^15.1.0) but use "unknown-options-as-args=true" yargs' option for impt commands (it fixes the issue). That option was not utilized by impt before.
This workaround is investigated and tested. No significant side effects found but few impt commands need to be updated to continue work correctly.

Our suggestion is the workaround 1)

@deldrid1 @mm-gmbd do you agree?

**Comment by mm-gmbd** [comment]
Workaround 1 sounds like the right path to me. Proceed!

**Comment by alexey-nbl** [comment]
New yargs version is released (15.2.0) that contains the new version of yargs-parser which should contain the fix for the issue we reported (https://github.com/yargs/yargs-parser/issues/232).
Need to carefully retest impt with the new yargs and decide what to do further.

**Comment by alexey-nbl** [comment]
Retested with the latest yargs/yargs-parser.
The issue has been fixed only partially. A new issue against yargs-parser is submitted - https://github.com/yargs/yargs-parser/issues/254

**Comment by alexey-nbl** [comment]
Workaround 1) is merged.
To track the status with the "yargs" module - https://github.com/EatonGMBD/imp-central-impt/issues/19 is submitted.

**Comment by mm-gmbd** [comment]
Fixed with #10
